src/main/java/com/minecolonies/entity/ai/citizen/farmer/EntityAIWorkFarmer.java

```diff
@@ -1,647 +1,561 @@
+            requestSeeds = false;
+            //Flag 1+2 is needed for updates
+            return true;
         }
         else
         {
-            if (plowedLand.isEmpty() && crops.isEmpty())
-            {
-                job.setStage(Stage.SEARCHING_LAND);
-            }
-            else
-            {
-                job.setStage(Stage.WORKING);
-            }
+            return false;
         }
     }
 
-    private int getFirstSeed()
+    /**
+     * Terminates the planting process and resets the task.
+     *
+     * @param buildingFarmer the building of the farmer.
+     * @param field          the field being planted.
+     * @return the next state.
+     */
+    private AIState terminatePlanting(BuildingFarmer buildingFarmer, Field field)
     {
-        for (int slot = 0; slot < getInventory().getSizeInventory(); slot++)
+        if (requestSeeds)
         {
-            ItemStack stack = getInventory().getStackInSlot(slot);
-
-            if (stack != null && stack.getItem() != null)
-            {
-                Item content = stack.getItem();
-                if (isSeed(content))
-                {
-                    return slot;
-                }
-            }
+            chatSpamFilter.talkWithoutSpam("entity.farmer.NeedSeed", field.getSeed().getItemStackDisplayName(new ItemStack(field.getSeed())));
         }
-        return -1;
+        else
+        {
+            resetVariables();
+            buildingFarmer.getCurrentField().setNeedsWork(false);
+            buildingFarmer.getCurrentField().setFieldStage(PLANTED);
+        }
+        return AIState.IDLE;
     }
 
-    private void harvesting()
+    /**
+     * Resets the basic variables of the class.
+     */
+    private void resetVariables()
     {
-        if (!crops.isEmpty())
-        {
-            delay = 10;
-
-            Block block = BlockPosUtil.getBlock(world, crops.get(0));
-
-            if (BlockPosUtil.isWorkerAtSiteWithMove(worker, crops.get(0)))
-            {
-                //todo should be age, may be different!
-                if (block == Blocks.melon_block || block == Blocks.pumpkin || world.getBlockState(crops.get(0)).getValue(BlockCrops.AGE)
-                                                                              == 0x7)
-                {
-                    List<ItemStack> items = BlockPosUtil.getBlockDrops(world, crops.get(0), 0);
+        requestSeeds = true;
+        shouldDumpInventory = true;
+        shouldTryToGetSeed = true;
+    }
 
-                    for (ItemStack item : items)
-                    {
-                        InventoryUtils.setStack(getInventory(), item);
-                    }
-                    //TODO add block break effect
-                    world.setBlockToAir(crops.get(0));
-                    harvestCounter++;
+    /**
+     * Executes the harvesting of the field.
+     *
+     * @return the next state.
+     */
+    private AIState harvest()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "harvest" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUnusedPrivateMethod)